fe/src/main/java/org/apache/doris/mysql/nio/ReadListener.java

```diff
@@ -0,0 +1,66 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package org.apache.doris.mysql.nio;
+
+import org.apache.doris.qe.ConnectContext;
+import org.apache.doris.qe.ConnectProcessor;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.xnio.ChannelListener;
+import org.xnio.XnioIoThread;
+import org.xnio.conduits.ConduitStreamSourceChannel;
+
+/**
+ * Created by infear on 2019/12/20.
+ */
+public class ReadListener implements ChannelListener<ConduitStreamSourceChannel> {
+    private final Logger LOG = LogManager.getLogger(this.getClass());
+    private NConnectContext ctx;
+    private ConnectProcessor connectProcessor;
+
+    public ReadListener(NConnectContext nConnectContext, ConnectProcessor connectProcessor) {
+        this.ctx = nConnectContext;
+        this.connectProcessor = connectProcessor;
+    }
+
+    @Override
+    public void handleEvent(ConduitStreamSourceChannel channel) {
+        // suspend must be call sync in current thread (the IO-Thread notify the read event),
+        // otherwise multi handler(task thread) would be waked up by once query.
+        XnioIoThread.requireCurrentThread();
+        ctx.suspendAcceptQuery();
+        // start async query handle in task thread.
+        channel.getWorker().execute(() -> {
```

Comment:
I don't know much about the use of `jboss.xnio` library, just read its documentation.

I have a question here: will this run in `WORKER THREAD pool`, which was configured to 16 in `NMysqlServer`(use `Options.WORKER_TASK_MAX_THREADS`)? 
If it is, there may be a problem. For example: if we have 16 slow queries, such as running for 5 minutes, which makes these 16 threads are filled. And because there are no threads available for subsequent query requests, the server will hang for 5 minutes.

Reply:
I have the same question of @lingbin . Because it's common that a query last for some minutes. If the worker pool size is static, it will cause other query can't be processed.

Reply:
@lingbin @imay 
After discussion with @imay, according to the case in doris, it woud be better to use a cahceThreadPool instead of fixed-size(16) thread pool mentioned above. I would optimize this part and also try to make users be able to inject customized thread pool. It should be more flexible.
Thanks for your advice!